### PR TITLE
pom file improvements

### PR DESCRIPTION
**Pull request by gotimer**
Added properties for dependency versions
Added properties for plugin versions
Added some more helpful properties
Defined minimum Maven version
Added a few plugins for security and compatibility checking

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -4,9 +4,38 @@
...
+    <prerequisites>
+        <maven>3.0.4</maven>
+    </prerequisites>
+
     <properties>
+        <java.version>1.8</java.version>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
         <suiteXmlFile>integration.xml</suiteXmlFile>
```
Comment (msaperst) [inline pom.xml]: as long as we're refactoring, one of the things I'd like to do is remove the need for this integration.xml file
We can put the information from this file directly into the failsafe configuration

Comment (msaperst) [inline pom.xml]: is there a maven naming convention that we are not following?

Comment (gotimer) [reply pom.xml]: Not sure which names are are asking about, but mostly no. There are some built-in parameters you weren't using at all (project.build.sourceEncoding) and some that you've created duplicates for (skipITs vs skip.integration.tests).

The lack of defined encodings was creating the "Build is platform-dependent" warnings during builds.

Comment (gotimer) [reply pom.xml]: I'll look at it next round as part of trying to self-contain the integration tests in Maven.

**Pull request merged**